packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/BaseScopeHandler.ts

```diff
@@ -106,15 +106,26 @@ export default abstract class BaseScopeHandler implements ScopeHandler {
       ...requirements,
     };
 
+    let currentPosition = position;
     for (const scope of this.generateScopeCandidates(
       editor,
       position,
       direction,
       hints,
     )) {
-      if (shouldYieldScope(position, direction, hints, previousScope, scope)) {
+      if (
+        shouldYieldScope(
+          currentPosition,
```

Comment:
Instead of determining whether to yield a scope based on where we started, we should more closely follow the description in our `ScopeHandler` interface and have a position that walks through the document.